service/src/main/java/bio/terra/cbas/runsets/results/RunResultsManager.java

```diff
@@ -0,0 +1,157 @@
+package bio.terra.cbas.runsets.results;
+
+import static bio.terra.cbas.common.MetricsUtil.recordMethodCompletion;
+
+import bio.terra.cbas.common.DateUtils;
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.models.CbasRunStatus;
+import bio.terra.cbas.models.Run;
+import bio.terra.cbas.runsets.monitoring.SmartRunsPoller;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RunResultsManager {
+  private final RunDao runDao;
+  private final SmartRunsPoller smartRunsPoller;
+  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(RunResultsManager.class);
+
+  public RunResultsManager(RunDao runDao, SmartRunsPoller smartRunsPoller) {
```

Comment:
I'm hoping with the change to receive failures directly we can remove this (circular?) dependency on a `SmartRunsPoller`